src/core/reactor.cc

```diff
@@ -2834,7 +2796,42 @@ class reactor::io_queue_submission_pollfn final : public reactor::pollfn {
         _armed = true;
         return true;
     }
-    virtual void exit_interrupt_mode() override final {
+public:
+    kernel_events_pollfn(reactor& r) : _r(r) {}
+    bool poll() final {
+        bool work = false;
+        work |= _r.reap_kernel_completions();
+        work |= _r.flush_pending_aio();
+#ifndef HAVE_OSV
+        work |= _r._backend->kernel_submit_work();
+#endif
+        work |= _r.reap_kernel_completions();
+        return work;
```

Comment:
We should merge these into a single call, but better to wait with that until later and proceed in small steps.